src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -949,4 +984,73 @@ mod tests {
             }
         }
     }
+
+    macro_rules! msrs_from_entries {
+        ($msr_entries:expr) => {
+            Msrs::from_entries(
+                &$msr_entries
+                    .iter()
+                    .map(|&(index, data)| kvm_msr_entry {
+                        index,
+                        data,
+                        ..Default::default()
+                    })
+                    .collect::<Vec<_>>(),
+            )
+            .unwrap()
+        };
+    }
+
+    macro_rules! assert_msrs {
+        ($msr_chunks:expr, $expected_msr_entries:expr) => {
+            let expected_entries: Vec<(u32, u64)> = $expected_msr_entries.iter().cloned().collect();
+            for msrs in $msr_chunks.iter() {
+                for &(index, data) in &expected_entries {
+                    if let Some(msr_data) = msrs
+                        .as_slice()
+                        .iter()
+                        .find(|msr| msr.index == index)
+                        .map(|msr| msr.data)
+                    {
+                        assert_eq!(msr_data, data);
+                    }
+                }
+            }
+        };
+    }
+
+    #[test]
+    fn test_fix_zero_tsc_deadline_msr_zero() {
+        let mut msr_chunks = vec![
+            msrs_from_entries!(&[(MSR_TSC_AUX, 1)]),
+            msrs_from_entries!(&[(MSR_IA32_TSC_DEADLINE, 0), (MSR_IA32_TSC, 42)]),
+        ];
+
+        KvmVcpu::fix_zero_tsc_deadline_msr(&mut msr_chunks);
+
+        // We expect for the MSR_IA32_TSC_DEADLINE to get updated with the MSR_IA32_TSC value, while
+        // other MSRs should remain unchanged.
+        assert_msrs!(
+            msr_chunks,
+            &[
+                (MSR_TSC_AUX, 1),
+                (MSR_IA32_TSC_DEADLINE, 42),
+                (MSR_IA32_TSC, 42)
+            ]
```

Comment:
Considering `msr_chunks` is a `Vec<Vec<...>>` to do an easier comparison you can flatten the vector into just `Vec<...>` and do a simple loop over 2 vectors.
```Rust
let flat_msr_chunks = msr_chunks.iter().flatten().cloned().collect();
let correct_msrs = [
                (MSR_TSC_AUX, 1),
                (MSR_IA32_TSC_DEADLINE, 42),
                (MSR_IA32_TSC, 42)
            ];
for (a, b) in flat_msr_chunks.iter().zip(correct_msrs.iter()) {
  assert_eq!(a.index, b.index);
  assert_eq!(a.data, b.data);
}
```

Reply:
done.